modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/extension/creole_dialog_definition.js

```diff
@@ -47,7 +47,11 @@ CKEDITOR.on('dialogDefinition', (event) => {
 			infoTab.remove('cmbWidthType');
 			infoTab.remove('cmbWidthType');
 			infoTab.remove('htmlHeightType');
-			infoTab.remove('txtBorder');
+
+			if (infoTab.get('txtBorder')) {
+				infoTab.remove('txtBorder');
+			}
```

Comment:
Naïve question: what happens if you try to remove a button that isn't there? ie. could this be just:

```js
infoTab.remove('txtBorder');
```

without the `if`?

Reply:
Well that doesn't change much.

There is still something strange about [this](https://github.com/liferay-frontend/liferay-portal/pull/234) bug, because the error only happens the first time you click on the **table** button, and that's because as noted [here](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR.html#event-dialogDefinition): 
> Note that this event is called only the first time a specific dialog is opened. Successive openings will use the cached dialog, and this event will not get fired.

What's even stranger is that I said that removing the `txtBorder` from the dialog fixes the problem because now in this branch with liferay-ckeditor version `4.14.1-liferay.6` and alloyeditor version `2.11.2` if keeps on crashing with the same error, so I'm going to keep on searching to see what's going on

Reply:
Since we want to do is avoid removing the `txtBorders` object from the dialog I'm going to reopen #234 and close this

Reply:
> Since we want to do is avoid removing the txtBorders object from the dialog I'm going to reopen #234 and close this

I don't understand this... we should be removing it, that's the whole point of this line of code, isn't it? We just need to be careful as to when are we removing it 🤔

Reply:
I know this is confusing:

First of all because my change is incorrect and I think should avoid removing the `txtBorder` field like @boton previously suggested in #234

We don't want to remove the `txtBorder` field from the info tab because, as I saw [here](https://github.com/ckeditor/ckeditor4/blob/eb2d11644a796cd13cd047cd55541430359f9317/plugins/showborders/plugin.js#L131-L132) the `showborders` plugin accesses the `txtBorder`field / object, so if it's not present that's when the error occurs.

Concerning:

> We just need to be careful as to when are we removing it 

As you've pointed out we're always removing it from `creole_dialog_definition.js`, which gets [included](https://github.com/liferay/liferay-portal/blob/f58ef75a756bf008ab5b2de50bcd2c295d04d89a/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/java/com/liferay/frontend/editor/ckeditor/web/internal/servlet/taglib/CKEditorCreoleOnEditorCreateDynamicInclude.java#L49-L51) by the `CKEditorCreoleOnEditorCreateDynamicInclude` class but I don't understand (yet) why this would be a problem in other situations.

Reply:
Just confirming what @boton already commented in #230's description

When the `showborders` plugin is bundled with `liferay-ckeditor`, when clicking on the table button, the `dialogDefinition` [event listener](https://github.com/ckeditor/ckeditor4/blob/eb2d11644a796cd13cd047cd55541430359f9317/plugins/showborders/plugin.js#L125) in the `showborders` plugin is invoked, it's able to access the `txtBorder` field from the info tab because it hasn't been removed yet. After that the `dialogDefintion` [event listener](https://github.com/liferay/liferay-portal/blob/f58ef75a756bf008ab5b2de50bcd2c295d04d89a/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/extension/creole_dialog_definition.js#L15) in `creole_dialog_defintion.js` is invoked and removes the `txtBorder` field from the info tab.

When the plugin is not bundled, which is the case right now, it's the other way around: the `dialogDefinition` [event listener](https://github.com/liferay/liferay-portal/blob/f58ef75a756bf008ab5b2de50bcd2c295d04d89a/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/extension/creole_dialog_definition.js#L15) in `creole_dialog_defintion.js`  is invoked, removes the `txtBorder` field and then the `dialogDefinition` [event listener](https://github.com/ckeditor/ckeditor4/blob/eb2d11644a796cd13cd047cd55541430359f9317/plugins/showborders/plugin.js#L125) in the `showborders`  plugin is invoked, tries to access the `txtBorder` field which has been removed and throws an error.

Reply:
> Just confirming what @boton already commented in #230's description
> 
> When the `showborders` plugin is bundled with `liferay-ckeditor`, when clicking on the table button, the `dialogDefinition` [event listener](https://github.com/ckeditor/ckeditor4/blob/eb2d11644a796cd13cd047cd55541430359f9317/plugins/showborders/plugin.js#L125) in the `showborders` plugin is invoked, it's able to access the `txtBorder` field from the info tab because it hasn't been removed yet. After that the `dialogDefintion` [event listener](https://github.com/liferay/liferay-portal/blob/f58ef75a756bf008ab5b2de50bcd2c295d04d89a/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/extension/creole_dialog_definition.js#L15) in `creole_dialog_defintion.js` is invoked and removes the `txtBorder` field from the info tab.
> 
> When the plugin is not bundled, which is the case right now, it's the other way around: the `dialogDefinition` [event listener](https://github.com/liferay/liferay-portal/blob/f58ef75a756bf008ab5b2de50bcd2c295d04d89a/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/extension/creole_dialog_definition.js#L15) in `creole_dialog_defintion.js` is invoked, removes the `txtBorder` field and then the `dialogDefinition` [event listener](https://github.com/ckeditor/ckeditor4/blob/eb2d11644a796cd13cd047cd55541430359f9317/plugins/showborders/plugin.js#L125) in the `showborders` plugin is invoked, tries to access the `txtBorder` field which has been removed and throws an error.

This is right, very detailed and accurate.

Reply:
So... what happens now in `creole`? Does the `txtBorder` field show when editing a table? What effect does it have when you set values to it?

Reply:
The `.on` method of `CKEDITOR` takes a [`priority`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#method-on) argument, so why not use it to reorder this and have a more predictable listener order instead?

Reply:
@jbalsas I hadn't seen it, but yes that seems like a good option thanks